File: week4/codigo_destino/temp_destino.cpp
#include <iostream>
#include <iomanip>

int main() {
    int iterations = 10'000'000;
    int param1 = 4;
    int param2 = 1;
    
    double result = 1.0;
    for (int i = 1; i <= iterations; ++i) {
        int j = i * param1 - param2;
        result -= (1.0 / j);
        j = i * param1 + param2;
        result += (1.0 / j);
    }
    
    result *= 4;
    
    std::cout << std::fixed << std::setprecision(12) << result << std::endl;
    
    return 0;
}

File: week4/codigo_destino/temp_destino.py
import numpy as np

def calculate(iterations, param1, param2):
    i = np.arange(1, iterations + 1, dtype=np.float64)
    j1 = i * param1 - param2
    j2 = i * param1 + param2
    result = 1.0 - np.sum(1.0 / j1) + np.sum(1.0 / j2)
    return result

result = calculate(10000000, 4.0, 1.0) * 4.0
print(f"{result:.12f}")